test/box-luatest/gh_6930_mvcc_isolation_levels_test.lua

```diff
@@ -0,0 +1,140 @@
+
+        local f = fiber.create(function()
+            fiber.self():set_joinable(true)
+            s:insert{1}
+        end)
+
+        t.assert_equals(s:select(), {})
+        t.assert_equals(s:count(), 0)
+
+        box.begin()
+        local res1 = s:select()
+        local res2 = s:count()
+        box.commit()
+        t.assert_equals(res1, {})
+        t.assert_equals(res2, 0)
+
+        local expect0 = {'default', 'read-confirmed', 'best-effort',
+                         box.txn_isolation['default'],
+                         box.txn_isolation['read-confirmed'],
+                         box.txn_isolation['best-effort']}
+
+        for _,level in pairs(expect0) do
+            box.begin{txn_isolation = level}
+            res1 = s:select()
+            res2 = s:count()
+            box.commit()
+            t.assert_equals(res1, {})
+            t.assert_equals(res2, 0)
+        end
+
+        expect0 = {'read-confirmed', 'best-effort',
+                    box.txn_isolation['read-confirmed'],
+                    box.txn_isolation['best-effort']}
+
+        for _,level in pairs(expect0) do
+            box.cfg{default_txn_isolation = level}
+            box.begin{}
+            res1 = s:select()
+            res2 = s:count()
+            box.commit()
+            t.assert_equals(res1, {})
+            t.assert_equals(res2, 0)
+            box.cfg{default_txn_isolation = 'best-effort'}
+        end
+
+        local expect1 = {'read-committed', box.txn_isolation['read-committed']}
+
+        for _,level in pairs(expect1) do
+            box.begin{txn_isolation = level}
+            res1 = s:select()
+            res2 = s:count()
+            box.commit()
+            t.assert_equals(res1, {{1}})
+            t.assert_equals(res2, 1)
+        end
+
+        for _,level in pairs(expect1) do
+            box.cfg{default_txn_isolation = level}
+            box.begin{txn_isolation = level}
+            res1 = s:select()
+            res2 = s:count()
+            box.commit()
+            t.assert_equals(res1, {{1}})
+            t.assert_equals(res2, 1)
+            -- default_txn_isolation does not affect autocommit select,
+            -- which is always run as read-confirmed
+            t.assert_equals(s:select(), {})
+            t.assert_equals(s:count(), 0)
+            box.cfg{default_txn_isolation = 'best-effort'}
+        end
+
+        -- With default best-effort isolation RO->RW transaction can be aborted:
+        box.begin()
+        res1 = s:select(1)
+        s:replace{2}
+        t.assert_error_msg_content_equals(
+            "Transaction has been aborted by conflict",
+            function() box.commit() end)
+        t.assert_equals(res1, {})
+
+        -- But using 'read-committed' allows to avoid conflict:
+        box.begin{txn_isolation = 'read-committed'}
+        res1 = s:select(1)
+        s:replace{2}
+        box.commit()
+        t.assert_equals(res1, {{1}})
+        t.assert_equals(s:select{}, {{1}, {2}})
```

Comment:
Nit: please leave a comment about why we expect to see this result: e.g., because of a series of fiber yields, both `inserts` become confirmed.